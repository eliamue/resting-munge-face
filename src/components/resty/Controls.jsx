import React from 'react';
import PropTypes from 'prop-types';

const Controls = ({ onChange, onSubmit, url, body }) => {
  return (
    <>
      <form 
        onSubmit={onSubmit} 
        onChange={onChange} 
        data-testid="controls-form"
      >
        <label htmlFor="url">URL</label>
        <input
          aria-label="url-input"
          type="url"
          name="url"
          id="url"
          value={url}
          placeholder="https://"
          onChange={onChange}
        />

        <section className="request-types">
          <label htmlFor="GET">GET</label>
          <input 
            type="radio"
            name="request"
            id="get" 
            onChange={onChange} 
          />
          <label htmlFor="PUT">PUT</label>
          <input 
            type="radio"
            name="request" 
            id="put" 
            onChange={onChange} 
          />
          <label htmlFor="POST">POST</label>
          <input 
            type="radio"
            name="request" 
            id="post" 
            onChange={onChange} 
          />
          <label htmlFor="DELETE">DELETE</label>
          <input 
            type="radio"
            name="request" 
            id="delete" 
            onChange={onChange} 
          />
        </section>

        <button aria-label="submit-request">Submit Request</button>
        
        <input
          aria-label="json-body"
          type="text"
          placeholder="JSON body"
          name="body"
          value={body}
          onChange={onChange}
        />
      </form>
    </>
  );
};

Controls.propTypes = {
  onChange: PropTypes.func.isRequired,
  onSubmit: PropTypes.func.isRequired,
  url: PropTypes.string.isRequired,
  body: PropTypes.string.isRequired,
};

export default Controls;
