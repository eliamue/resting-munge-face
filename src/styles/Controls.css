body {
    justify-content: space-around;
    justify-self: space-around;
    justify-items: space-around;
    text-align: center;
    background-color: rgb(224, 230, 223);
    margin-left: 150px;
}

textarea {
    display: flex;
    text-align: flex-start;
    height: 50px;
    width: 400px;
}

label {
    justify-content: space-between;
    justify-self: space-between;
    justify-items: space-between;
    text-align: space-between;
    padding: 15px;
}

form {
    display: flex;
    flex-direction: column;
    width: 100%;
    margin-left: 150px;
    margin-top: 10px;
    align-items: flex-start;
    justify-content: space-around;
}

button {
    width: fit-content;
}

.url {
    width: 80%;
}
