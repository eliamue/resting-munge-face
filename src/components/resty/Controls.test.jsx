import React from 'react';
import { render, screen } from '@testing-library/react';
import Controls from './Controls';

describe('Controls', () => {
  it('displays controls that include radio buttons for each route, submit button and text box', () => {
    render(<Controls 
      url="" 
      body="" 
      onChange={() => {}} 
      onSubmit={() => {}} 
    />);
    const controls = screen.getByTestId('controls-form');
    expect(controls).toMatchSnapshot();
  });
});
