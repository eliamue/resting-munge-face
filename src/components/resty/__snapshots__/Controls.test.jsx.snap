// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Controls displays controls that include radio buttons for each route, submit button and text box 1`] = `
<form
  data-testid="controls-form"
>
  <label
    for="url"
  >
    URL
  </label>
  <input
    aria-label="url"
    id="url"
    name="url"
    placeholder="https://"
    type="url"
    value=""
  />
  <section
    class="route-types"
  >
    <label
      for="GET"
    >
      GET
    </label>
    <input
      aria-label="GET"
      id="get"
      name="route"
      type="radio"
      value="GET"
    />
    <label
      for="PUT"
    >
      PUT
    </label>
    <input
      id="put"
      name="route"
      type="radio"
      value="PUT"
    />
    <label
      for="POST"
    >
      POST
    </label>
    <input
      id="post"
      name="route"
      type="radio"
      value="POST"
    />
    <label
      for="DELETE"
    >
      DELETE
    </label>
    <input
      id="delete"
      name="route"
      type="radio"
      value="DELETE"
    />
  </section>
  <button
    aria-label="submit-request"
  >
    Submit Request
  </button>
  <textarea
    aria-label="body"
    name="body"
    placeholder="JSON body"
    type="text"
  />
</form>
`;
